### PR TITLE
Recents

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## What changed (plus any additional context for devs)
- Clean up unused BrowserState store
- Fix race conditions while extracting icon metadata
- Store last 1k visited websites with metadata and timestamp
- Show the last 10 recent ones in the new tab page.
- Shows nothing if no recents.

## Screen recordings / screenshots
Screenshot 2024-04-12 at 2 03 18 PM

### CODE REVIEW

File: src/state/browserState/index.ts [hunk 1]
```diff
@@ -1,67 +0,0 @@
-import create from 'zustand';
```
Comment (skylarbarrera) [inline src/state/browserState/index.ts:1]: gg

**Review by skylarbarrera — APPROVED** [review]
code good

File: src/components/DappBrowser/Homepage.tsx [hunk 2]
```diff
@@ -114,11 +128,11 @@ const Card = ({ site, showMenuButton }: { showMenuButton?: boolean; site: Trendi
...
               </Box>
               <Stack space="10px">
-                <Text size="17pt" weight="heavy" color="label">
+                <Text size="17pt" weight="heavy" color="label" numberOfLines={2}>
                   {site.name}
                 </Text>
-                <Text size="13pt" weight="bold" color="labelTertiary">
-                  {site.url}
+                <Text size="13pt" weight="bold" color="labelTertiary" numberOfLines={1}>
+                  {site.url.startsWith('http:') || site.url.startsWith('https') ? getDappHost(site.url) : site.url}
```
Comment (benisgold) [inline src/components/DappBrowser/Homepage.tsx:135]: could just do `site.url.startsWith('http')` right?

**Review by benisgold — APPROVED** [review]

File: src/components/DappBrowser/Homepage.tsx (hunk 2 above)
Comment (brunobar79) [reply src/components/DappBrowser/Homepage.tsx:135]: `httpwebsite.com`  is a valid domain
```suggestion
                  {site.url.startsWith('http:') || site.url.startsWith('https:') ? getDappHost(site.url) : site.url}
```

**Comment by brunobar79** [comment]
Merging anyways,  e2e failing is unrelated

**Pull request merged**